timescaledb/how-to-guides/backup-and-restore/pg-dump-and-restore.md

```diff
@@ -1,114 +1,103 @@
 
 ```bash
-pg_dump -Fc -f tutorial.bak tutorial
-```
-
-Restoring data from a backup currently requires some additional procedures,
-which need to be run from `psql`:
-```sql
-CREATE DATABASE tutorial;
-\c tutorial --connect to the db where we'll perform the restore
-CREATE EXTENSION IF NOT EXISTS timescaledb;
-SELECT timescaledb_pre_restore();
-
--- execute the restore (or from a shell)
-\! pg_restore -Fc -d tutorial tutorial.bak
-
-
-SELECT timescaledb_post_restore();
+pg_dump -Fc -f exampledb.bak exampledb
 ```
 
-<highlight type="warning"> PostgreSQL's `pg_dump` does not currently specify the *version* of
- the extension in its backup, which leads to problems if you are
- restoring into a database instance with a more recent extension
- version installed.  (In particular, the backup could be for some
- version 1.6, but then the `CREATE EXTENSION timescaledb` command just
- installs the latest (say, 1.7), and thus does not have the
- opportunity to run our upgrade scripts.)  
-
-The workaround is that when restoring from a backup, you need to
- restore to a PostgreSQL instance with the same extension version
- installed, and *then* upgrade the version.
- </highlight>
-
 <highlight type="warning">
- These instructions do not work if you use flags to selectively
- choose tables (`-t`) or schemas (`--schema`), and so cannot be used
- to backup only an individual hypertable.  In particular, even if you
- explicitly specify both the hypertable and all of its constituent
- chunks, this dump would still lack necessary information that
- TimescaleDB stores in the database catalog about the relation between
- these tables.
-
-You can, however, explicitly *exclude* tables from this whole
- database backup (`-T`), as well as continue to selectively backup
- plain tables (i.e., non-hypertable) as well.
-</highlight>
-
-
-## Individual hypertables
-
-Below is the procedure for performing a backup and restore of hypertable `conditions`.
-
-### Backup
-
-Backup the hypertable schema:
-```bash
-pg_dump -s -d old_db --table conditions -N _timescaledb_internal | \
-  grep -v _timescaledb_internal > schema.sql
-```
-
-Backup the hypertable data to a CSV file.
-```bash
-psql -d old_db \
--c "\COPY (SELECT * FROM conditions) TO data.csv DELIMITER ',' CSV"
-```
-
-### Restore
-Restore the schema:
-```bash
-psql -d new_db < schema.sql
-```
-
-Recreate the hypertables:
-```bash
-psql -d new_db -c "SELECT create_hypertable('conditions', 'time')"
-```
-
-<highlight type="tip">
-The parameters to `create_hypertable` do not need to be
-the same as in the old db, so this is a good way to re-organize
-your hypertables (e.g., change partitioning key, number of
-partitions, chunk interval sizes, etc.).
+Do not use the `pg_dump` command to backup individual hypertables. Dumps created using this method lack the necessary information to correctly restore the hypertable from backup.
```

Comment:
This simplifies the content on the left quite a bit, which isn't bad, but it does feel like we're leaving something out here. In essence, the previous content was saying, "if you backup and restore TimescaleDB with `pg_dump`, it must always be restored to a PostgreSQL database running the **same** version of TimescaleDB before upgrading to a newer release of the extension."

Is it worth adding something like this in here?

Reply:
I say that in more detail under the Troubleshooting heading. Adding that info here seems like we'd be conflating two different ideas. Here in the admon, we have the idea of backing up individual hypertables, where later on we deal with the idea of version mismatches.

Reply:
Maybe my overall point is that separating these two actions means that somone could theoretically read the section on backing up, do that for some period of time without restoring (bad practice, I know... but...) and then when they are in trouble and try to restore, it doesn't work and they're left in sticky situation. In truth, it's not the biggest, biggest deal, but _maybe_ if we said upfront that, "hey, you need to always keep track of what Postgres and TimescaleDB version you're on in case you need to restore to the exact same version..." - folks would be more inclined to keep track fo that?

I mean, in theory it's a moot point with David's tool and `pg_dump` isn't/shouldn't be your main backup mechanism, but that's where my thinking is at with it.

Reply:
OK, that's a fair point. I'll have a play with it and see what I can come up with.